h2o-core/src/main/java/water/ScopeInspect.java

```diff
@@ -0,0 +1,125 @@
+package water;
+
+import water.fvec.Frame;
+import water.fvec.Vec;
+
+import java.util.*;
+import java.util.function.Predicate;
+
+/**
+ * Some utility functions useful for debugging when trying to find origin of leaked keys.
+ */
+public class ScopeInspect {
```

Comment:
those printing functions helped me a lot to track leaked keys, so I'm proposing to keep them in this dedicated class.